contracts/mechanisms/LockDrop.sol

```diff
@@ -1,36 +1,92 @@
 pragma solidity ^0.4.24;
 
 import "openzeppelin-zos/contracts/math/SafeMath.sol";
+import "../token/EdgewareERC20.sol";
 
 /**
  * The LockDrop contract locks up funds of depositors
  * with a verifiable receipt/proof that can be used to
  * redeem tokens on a separate, compatible blockchain.
- * THe tokens with be locked for a specified time length.
+ * The tokens with be locked for a specified time length.
+ * 
+ * The tokens are minted at a rate that grows according
+ * to the chosen locking schedule.
  */
 contract LockDrop {
-    uint public lockDropCapacity;
-    uint public lockDropBeginning;
-    uint public lockDropEnding;
+    uint public capacity;
+    uint public beginning;
+    uint public ending;
+    uint public maxLength;
 
-    mapping (address => uint) lockDropBalances;
+    struct Deposit {
+        uint amount;
+        uint lockEnding;
+    }
+    
+    mapping (address => Deposit[]) deposits;
 
-    constructor(uint length, uint capacity) {
-        lockDropCapacity = capacity;
-        lockDropBeginning = now;
-        lockDropEnding = now * length * 1 day;
+    constructor(uint _lockPeriod, uint _maxLength, uint _capacity) {
+        beginning = now;
+        ending = now + (1 days * _lockPeriod);
+        capacity = _capacity;
+        maxLength = _maxLength;
     }
 
-    function lock() payable public {
-        require(now <= lockDropEnding);
-        uint balance = lockDropBalances[msg.sender]
-        lockDropBalances[msg.sender] = SafeMath.add(balance, msg.value);
+    /**
+     * @dev        Lock function for participating in the lock drop
+     * @param      _length  The length of time chosen for locking ether
+     */
+    function lock(uint _length) payable public {
+        require(now <= ending);
+
+        // Create deposit with paid amount and specified length
+        // of time. The lock ending is determined as the specified
+        // time length after the ending of the lock up period in days.
+        Deposit d = Deposit({
+            amount: msg.value,
+            lockEnding: ending + (_length * 1 days)
+        });
+
```

Comment:
Incrementally add the `EffectiveAmount` to the `public uint TotalEffectiveDeposits` a new param, and `public uint TotalDeposits` as well

Reply:
I don't think I follow this tbh, if you are proposing new parameters please explain them. The `lockEnding` is just a time that this specific lock is up. It is decided after the ending as you can see.

Reply:
I'm guessing effectiveAmount is just the deposit.amount parameter.

So we have some `capacity` of ETH that we are taking, we decide this up front so this keeps track of how much `capacity` is left to lock up. If you lock ETH the `capacity` goes down, if you unlock it goes up.

After lock period has ended, you can withdraw only if the `lockEnding` time is over after the `ending` of the lock period.